### PR TITLE
@/climatemappedafrica `TreeView`

### LINKED ISSUE #925: ClimateMapped Data Navigation Panel
Create a navigation panel on Climatemapped Africa Similar to PesaYetu

Request Link: [Feature 5](https://docs.google.com/document/d/1TUrdCRnbj6TSqiuwITKUpnDUEi_ktm_3PmgsaD0y2bw/edit#heading=h.hbi7d1sv7wym)

> Preview
Image

### PR DESCRIPTION
**Pull request by kilemensi**
## Description

This PR implements/fixes the TreeView in explore pages.

Fixes #925 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Screenshots

Screenshot 2024-10-20 at 09 21 03

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: apps/climatemappedafrica/src/components/HURUmap/TreeView/TreeView.js [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+  background: "inherit",
+  borderRadius: 0,
+  borderBottom: `1px solid transparent`,
+  borderRight: `2px solid transparent`,
+  padding: 0,
+  "&:hover": {
+    background: "inherit",
+  },
+  "&.expanded": {
+    borderRadius: 0,
```
Comment (m453h) [inline apps/climatemappedafrica/src/components/HURUmap/TreeView/TreeView.js:32]: Isn't this also redundant ?
```suggestion
```

File: apps/climatemappedafrica/src/components/HURUmap/TreeView/TreeView.js [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+const CustomTreeItemContent = styled(TreeItem2Content)(({ theme }) => ({
+  background: "inherit",
+  borderRadius: 0,
+  borderBottom: `1px solid transparent`,
+  borderRight: `2px solid transparent`,
+  padding: 0,
+  "&:hover": {
+    background: "inherit",
+  },
```
Comment (m453h) [inline apps/climatemappedafrica/src/components/HURUmap/TreeView/TreeView.js:30]: I think we don't really need this right as its the same style ?

**Review by m453h — COMMENTED** [review]
🚀  Looks good

I hadn't looked at [TreeItem2](https://mui.com/x/api/tree-view/tree-item-2/) before, it does give a lot of customization options

Just a few things missing to make it exactly as the component we have on PesaYetu:

1. The background of the child items should change when on hover 
`ClimateMappedAfrica`
climatemapped
`PesaYetu`
  pesayetu

2. Only one Parent TreeView item should be expanded at a time:
`ClimateMappedAfrica`
ezgif-3-21299f0616
`PesaYetu`
ezgif-3-2a8c00ee56

File: apps/climatemappedafrica/src/components/HURUmap/TreeView/TreeView.js (hunk 1 above)
Comment (kilemensi) [reply apps/climatemappedafrica/src/components/HURUmap/TreeView/TreeView.js:32]: Let me test that locally and confirm...

**Comment by kilemensi** [comment]
Good catch on the hover @m453h 

>Only one Parent TreeView item should be expanded at a time:
>

I think this was a mistake in PesaYetu or at least I don't see why/how it helps users; if a user wants to compare what is in **Education** vs what is in **Health**, they **CAN'T** in PesaYetu design. You can't even say it's a screen realestate issue because the `TreeView` has the length of the page to expand.

File: apps/climatemappedafrica/src/components/HURUmap/TreeView/TreeView.js (hunk 2 above)
Comment (kilemensi) [reply apps/climatemappedafrica/src/components/HURUmap/TreeView/TreeView.js:30]: This is needed. Without it, 👇🏽 

https://github.com/user-attachments/assets/cebcf516-3347-4d60-a003-8ccb7679c45f

File: apps/climatemappedafrica/src/components/HURUmap/TreeView/TreeView.js (hunk 1 above)
Comment (kilemensi) [reply apps/climatemappedafrica/src/components/HURUmap/TreeView/TreeView.js:32]: Looks like this is not needed.

File: apps/climatemappedafrica/src/components/HURUmap/TreeView/TreeView.js (hunk 2 above)
Comment (m453h) [reply apps/climatemappedafrica/src/components/HURUmap/TreeView/TreeView.js:31]: Anha, I see 🚀

**Review by m453h — APPROVED** [review]
Looks good  🚀

**Pull request merged**